packages/js/admin-layout/src/components/LayoutContext/layout-context.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { createElement, createContext, useContext } from '@wordpress/element';
+
+export type LayoutContextType = {
+	layoutString: string;
+	updateLayoutPath: ( item: string ) => LayoutContextType;
+	layoutPath: string[];
```

Comment:
This and `layoutString` took me a minute to parse.  I wonder if something like `layoutParts`, `layoutBreadcrumbs` or `layoutHierarchy` might make this more clear.  (I was thinking `path` as in a string portion of the URL).

Similarly for `layoutString` this could just be `layout` if we make the above change.

Reply:
Thanks @joshuatf , I made the following changes:
```
layoutPath -> layoutParts
updateLayoutPath -> extendLayout
descendantOf -> isDescendantOf
```

I left `layoutString` as it is for now, as I feel that `layout` doesn't quite work for it in my mind, and I'd like the name to be a bit more unique that just `layout`. Definitely open to other suggests if you feel the need.